### ISSUE TITLE
Proposal: set app theme at runtime

### ISSUE THREAD

**Issue by niels9001** [microsoft/microsoft-ui-xaml#2991]
# Proposal: Support for changing the app's theme at runtime, not just start.

```
App.Current.RequestedTheme
```

## Rationale
It's currently only possible to update the App's theme in the constructor of App.xaml.cs, a.k.a. when the app starts. Would be great to have to support to do this at all time. I know you can set the requested theme of a piece of UI (like a page) but you still rin into minor issues sometimes (e.g. window buttons need to be updated as well). Setting the App's theme would resolve this.

**Comment by StephenLPeters** [comment]
@kikisaints This seems themeing related

**Comment by tsvietOK** [comment]
@niels9001 You could check how changing theme at runtime has been implemented in the source code of [Xaml-Controls-Gallery](https://github.com/microsoft/Xaml-Controls-Gallery)

**Comment by kikisaints** [comment]
You can achieve a result like this by naming your root mainpage and flipping the requested theme there back'n forth (light-dark-light or dark-light-dark) and that should cascade down into all the controls that are children within that page - however, this method/hack falls apart a bit with menus and flyouts as they are out of that "root" tree and don't receive the requested theme toggle/flip.

I understand niels' ask here, it's something I'd like too, I think it might be a bit more work than it appears on the surface.

**Comment by DHowett** [comment]
Terminal is interested in this as well. We have had to do some pretty unfortunate things to set our app's Theme ☹️

**Comment by niels9001** [comment]
Right, you get so far with setting the RequestedTheme on e.g. a page. Stuff like tooltip fall through though. Here's an example of the PowerToys settings:

![App xaml](https://user-images.githubusercontent.com/9866362/89162748-cb2f0380-d574-11ea-8bc0-926bbe814b25.gif)

Both Flyouts do not adhere correctly to the theme.

**Comment by castorix** [comment]
I still use Windows App SDK 1.6.241114003 and it seems to work correctly by using for example _ApplicationPageBackgroundThemeBrush_ for the parent control (tested wih a Grid)
When I switch Dark/Light theme in Windows 10 Settings, the theme is changed automatically for all child controls and Tooltips/TeachingTips/...
